bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java

```diff
@@ -680,22 +681,47 @@ public static BXml createXmlProcessingInstruction(BString target, BString data,
      *
      * @param mapType map type.
      * @return map value
+     * @deprecated use {@link #createMapValue(MapType)} instead
      */
+    @Deprecated
     public static BMap<BString, Object> createMapValue(Type mapType) {
         return new MapValueImpl<>(mapType);
     }
 
+    /**
+     * Create a runtime map value with given type.
+     *
+     * @param mapType map type.
+     * @return map value
+     */
+    public static BMap<BString, Object> createMapValue(MapType mapType) {
+        return new MapValueImpl<>(mapType);
+    }
+
     /**
      * Create a runtime map value with given initial values and given type.
      *
      * @param mapType   map type.
      * @param keyValues initial map values to be populated.
      * @return map value
+     * @deprecated use {@link #createMapValue(MapType, BMapInitialValueEntry[])} instead
      */
+    @Deprecated
     public static BMap<BString, Object> createMapValue(Type mapType, BMapInitialValueEntry[] keyValues) {
         return new MapValueImpl<>(mapType, keyValues);
     }
 
+    /**
+     * Create a runtime map value with given initial values and given type.
+     *
+     * @param mapType   map type.
+     * @param keyValues initial map values to be populated.
+     * @return map value
+     */
+    public static BMap<BString, Object> createMapValue(MapType mapType, BMapInitialValueEntry[] keyValues) {
+        return new MapValueImpl<>(mapType, keyValues);
+    }
+
```

Comment:
Shall we add some test cases with this API changes?

Reply:
Added tests for newly introduced 4 methods. Will add for other uncovered api methods also.